src/cluster_legacy.c

```diff
@@ -2815,6 +2869,61 @@ static clusterNode *getNodeFromLinkAndMsg(clusterLink *link, clusterMsg *hdr) {
     return sender;
 }
 
+static clusterMsgDataPublishMessage *getFirstPublishMessage(clusterMsgDataPublishMulti *msg) {
+    clusterMsgDataPublishMessage *initial = (clusterMsgDataPublishMessage *)&(msg->bulk_data);
+    return initial;
```

Comment:
This ref and casting seem wrong - isn't `bulk_data` already `clusterMsgDataPublishMessage *`? 

also nit - this seems quite verbose IMO

```suggestion
    return msg->bulk_data;
```

Reply:
We have removed the Multi publish struct for this PR